### PR TITLE
Add Japanese language

### PR DESCRIPTION
**Pull request by kudtomoy**
Fixes # 

---

- The English content has been moved to `workshop/content/english` and the Japanese content has been added as `workshop/content/japanese`.
- A button to switch languages has also been added.

* Please squash when merging.

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Comment by kellertk** [comment]
Hello @kudtomoy, thank you very much for your hard work on this! I noticed a few issues:

* http://localhost:1313/ja/50-java.html: The title of this page is "Java Workshop". Should it be "Java ワークショップ" instead?
* I can't switch languages on some pages, even though there is a translated page available. On http://localhost:1313/70-construct-hub.html I cannot select 日本語, but on http://localhost:1313/ja/60-construct-hub.html I cannot select English.
* We have added the Go workshop since you started this PR. It's OK to not have a translated version, but that means if I switch languages the menu on the left is no longer the same as the English page, and the numbering on the HTML pages don't match between 日本語 and English. We need to make sure the menu items and HTML numbering is consistent between languages.

Overall, this looks good aside from those few issues, and I think the language dropdown is great! Please correct the above items so we can move forward and merge this.

**Review by kellertk — CHANGES_REQUESTED** [review]
Hello @kudtomoy, thank you very much for your hard work on this! I noticed a few issues:

http://localhost:1313/ja/50-java.html: The title of this page is "Java Workshop". Should it be "Java ワークショップ" instead?
I can't switch languages on some pages, even though there is a translated page available. On http://localhost:1313/70-construct-hub.html I cannot select 日本語, but on http://localhost:1313/ja/60-construct-hub.html I cannot select English.
We have added the Go workshop since you started this PR. It's OK to not have a translated version, but that means if I switch languages the menu on the left is no longer the same as the English page, and the numbering on the HTML pages don't match between 日本語 and English. We need to make sure the menu items and HTML numbering is consistent between languages.
Overall, this looks good aside from those few issues, and I think the language dropdown is great! Please correct the above items so we can move forward and merge this.

File: workshop/content/english/40-dotnet/20-create-project/500-deploy.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ IAM Statement Changes
 Do you wish to deploy these changes (y/n)?
 ```
 
-This is warning you that deploying the app contains security-sensitive changes.
+This is warning you that deploying the app entails some risk.
```
Comment (peterwoodworth) [inline workshop/content/english/40-dotnet/20-create-project/500-deploy.md:62]: Please keep this text how it was 🙂

File: workshop/content/english/40-dotnet/60-cleanups/_index.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
++++
```
Comment (peterwoodworth) [inline workshop/content/english/40-dotnet/60-cleanups/_index.md:1]: Please ensure this file is up to date, a whole section was added to this page

File: workshop/content/english/40-dotnet/70-advanced-topics/100-pipelines/5000-test-actions.md [hunk 3]
```diff
@@ -153,7 +153,7 @@ namespace CdkWorkshop
 {{</highlight>}}
 
 Now we can add those values to our actions in `CdkWorkshop/PipelineStack.cs` by getting the `CfnOutput` of our deploy stage:
-{{<highlight ts "hl_lines=5 11">}}
+{{<highlight ts "hl_lines=6 15">}}
```
Comment (peterwoodworth) [inline workshop/content/english/40-dotnet/70-advanced-topics/100-pipelines/5000-test-actions.md:156]: These should be highlighting lines 5 and 11

File: workshop/content/english/40-dotnet/_index.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
++++
```
Comment (peterwoodworth) [inline workshop/content/english/40-dotnet/_index.md:1]: Please ensure this file is up to date - there should be a warning talking about cost

File: workshop/content/english/50-java/20-create-project/500-deploy.md [hunk 5]
```diff
@@ -60,7 +60,7 @@ IAM Statement Changes
 Do you wish to deploy these changes (y/n)?
 ```
 
-This is warning you that deploying the app contains security-sensitive changes.
+This is warning you that deploying the app entails some risk.
```
Comment (peterwoodworth) [inline workshop/content/english/50-java/20-create-project/500-deploy.md:63]: Please keep the text how it was 🙂

File: workshop/content/english/50-java/60-cleanups/_index.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
++++
```
Comment (peterwoodworth) [inline workshop/content/english/50-java/60-cleanups/_index.md:1]: Please make sure this file is up to date, there should be a whole section on resources which are retained by default

File: workshop/content/english/50-java/70-advanced-topics/100-pipelines/4000-build-stage.md [hunk 7]
```diff
@@ -82,7 +82,7 @@ public class WorkshopPipelineStack extends Stack {
 }
 {{</highlight>}}
 
-This imports and creates an instance of the `WorkshopPipelineStage`. Later, you might instantiate this stage multiple times (e.g. you want a Production deployment and a separate development/test deployment).
+This imports and creates an instance of the `WorkshopPipelineStage`. Later, you might instantiate this stage multiple times (e.g. you want a Production deployment and a separate devlopment/test deployment).
```
Comment (peterwoodworth) [inline workshop/content/english/50-java/70-advanced-topics/100-pipelines/4000-build-stage.md:85]: Please revert this typo

File: workshop/content/english/50-java/_index.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
++++
```
Comment (peterwoodworth) [inline workshop/content/english/50-java/_index.md:1]: Please make sure this file is up to date, there should be a note on cost

**Review by peterwoodworth — CHANGES_REQUESTED** [review]
This is great work, thanks a ton! Most of the changes I'm requesting are changes that have gone through likely since you've started working on this, so for most of these just make sure you're fully up to date with the repo!

Additionally, you can delete `clib.png` in all the instances where it's being added and uploaded since it is no longer being used

File: workshop/content/japanese/20-typescript/30-hello-cdk/200-lambda.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+
+また、もし __git__ を使用する場合は、`.gitignore` ファイルに `！lambda/*.js` を追加することを検討してください。
+
+## AWS Lambda コンストラクトライブラリをインストールする
+
+AWS CDK には、__AWS Construct Library__ と呼ばれるコンストラクトのための広範なライブラリが付属しています。AWS Construct Library は、AWS のサービスごとに独立した __モジュール__ として提供されます。たとえば、AWS Lambda 関数を定義する場合、AWS Lambda の Construct Library を使用する必要があります。
+
+AWS コンストラクトについて学ぶには、[AWS Construct Library reference](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-construct-library.html) を参照してください。
+
+![](./clib.png)
```
[the remark this answers is not in the document]
Comment (kudtomoy) [reply workshop/content/japanese/20-typescript/30-hello-cdk/200-lambda.md:40]: OK, in addition to TypeScript, the following pages were included and have been fixed.
```
$ find . -name "*.md" | xargs grep clib.png                                                                                                           ✘ 1
./content/japanese/50-java/30-hello-cdk/200-lambda.md:![](./clib.png)
./content/japanese/30-python/30-hello-cdk/200-lambda.md:![](./clib.png)
```

File: workshop/content/english/50-java/60-cleanups/_index.md (hunk 6 above)
Comment (kudtomoy) [reply workshop/content/english/50-java/60-cleanups/_index.md:1]: The English version has been updated and the Japanese version has been fixed.

**Comment by kudtomoy** [comment]
@peterwoodworth 
Thanks so much for the review, and your kind comments!
I have corrected all the points you raised.
Please let me know if additional work is needed.

File: workshop/content/japanese/30-python/50-table-viewer/200-install.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
++++
+title = "ライブラリのインストール"
+weight = 200
++++
+
+## pip install
```
Comment (peterwoodworth) [inline workshop/content/japanese/30-python/50-table-viewer/200-install.md:6]: Please update this page to be consistent with the english version. A snippet has been added and the content has been slightly modified to make things more clear

File: workshop/content/japanese/30-python/60-cleanups/_index.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
++++
+title = "クリーンアップ"
+weight = 60
+chapter = true
++++
+
+# スタックのクリーンアップ
```
Comment (peterwoodworth) [inline workshop/content/japanese/30-python/60-cleanups/_index.md:7]: Please update this cleanup file to be consistent with english version. Like the Typescript cleanup page in Japanese, the section on removal policy is missing 🙂

File: workshop/content/japanese/30-python/70-advanced-topics/200-pipelines/1000-setting-up.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+実際の「本番」アプリケーションとは分離するので、完全に独立させる必要があります。
+
+`cdk_workshop` フォルダの中で `pipeline_stack.py` という新しいファイルを作成します。ファイルに以下の内容を追加します。
+
+{{<highlight python>}}
+from constructs import Construct
+from aws_cdk import (
+    Stack
+)
+from pipeline_stage import WorkshopPipelineStage
```
Comment (peterwoodworth) [inline workshop/content/japanese/30-python/70-advanced-topics/200-pipelines/1000-setting-up.md:19]: I don't think the line `from pipeline_stage import WorkshopPipelineStage` is supposed to be here quite yet 🙂

File: workshop/content/japanese/30-python/70-advanced-topics/200-pipelines/3000-new-pipeline.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+`pipeline_stack.py` を開き、以下のように編集します。
+
+{{<highlight python "hl_lines=13 18-30">}}
+from constructs import Construct
+from aws_cdk import (
+    Stack,
+    aws_codecommit as codecommit,
+    pipelines as pipelines,
+)
+from cdk_workshop.pipeline_stage import WorkshopPipelineStage
```
Comment (peterwoodworth) [inline workshop/content/japanese/30-python/70-advanced-topics/200-pipelines/3000-new-pipeline.md:19]: I still don't think this is supposed to be here just yet 🙂 `from pipeline_stage import WorkshopPipelineStage`

File: workshop/content/japanese/20-typescript/_index.md [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+title = "TypeScript ワークショップ"
+chapter = true
+weight = 20
++++
+
+# TypeScript ワークショップ
+
+こちらのワークショップのバリエーションでは、TypeScript 言語で体験できます。
+
+
```
Comment (peterwoodworth) [inline workshop/content/japanese/20-typescript/_index.md:11]: Can we add the disclaimer about cost here?

File: workshop/content/japanese/30-python/_index.md [hunk 15]
```diff
@@ -0,0 +1,9 @@
++++
+title = "Python ワークショップ"
+chapter = true
+weight = 30
++++
+
+# Python ワークショップ
+
+こちらのワークショップのバリエーションでは、Python 言語で体験できます。
```
Comment (peterwoodworth) [inline workshop/content/japanese/30-python/_index.md:9]: Can we add the disclaimer about cost here?

**Review by peterwoodworth — CHANGES_REQUESTED** [review]
Ok I've fully reviewed everything until the .NET workshop starts. I will review the .NET workshop and Java workshop tomorrow. For now, here's everything else I've found:

Can we add a translation for [this page](https://cdkworkshop.com/15-prerequisites/900-go.html) describing how to install Go?

It looks like the Typescript and Python workshops are lacking `A disclaimer about cost` section in the Japanese workshop, be sure to add this note in both of these workshops!

On the [cleanup page for typescript](https://cdkworkshop.com/20-typescript/60-cleanups.html), it seems you copied the page from the Java section in the english version. Could you ensure you are using TypeScript here? Additionally in the Japanese version of this page, could you update it with the removal policy block I'm describing to be consistent with english version and latest useful info?

File: workshop/content/japanese/40-dotnet/70-advanced-topics/100-pipelines/4000-build-stage.md [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+
+## コミット/デプロイ
+アプリケーションをデプロイするためのコードを追加できたので、後は変更をコミットしてリポジトリにプッシュするだけです。
+
+
+```
+git commit -am "Add deploy stage to pipeline" && git push
+```
+
+プッシュが完了したら、[CodePipeline コンソール](https://console.aws.amazon.com/codesuite/codepipeline/pipelines) でパイプラインの実行状況を確認できます (しばらく時間がかかる場合があります)。
```
Comment (peterwoodworth) [inline workshop/content/japanese/40-dotnet/70-advanced-topics/100-pipelines/4000-build-stage.md:93]: Can we add these lines to the end of the file?

```
![](./pipeline-succeed.png)

成功しました !
```

**Review by peterwoodworth — CHANGES_REQUESTED** [review]
Looked over the Java and .NET workshops and gave my suggested fixes.

One additional thing, in each of the english workshops could you re-implement [this fix?](https://github.com/aws-samples/aws-cdk-intro-workshop/pull/660) This bug is still present in this PR since the linked PR was just recently merged

**Pull request merged**